modules/mobile/app/models/mobile/v0/adapters/immunizations.rb

```diff
@@ -33,8 +33,10 @@ def sort_by_date_and_group_name(vaccine_map)
           vaccine_map.sort_by do |immunization|
             date_sort_key1 = immunization[:date] ? 0 : 1 # used to keep nil dates at end of list
             date_sort_key2 = immunization[:date] ? -immunization[:date].to_i : nil
```

Comment:
I actually don't think the two date sort keys are necessary because it's just a numerical comparison and 0 is the highest number possible here. I think this would work as:
`date_sort_key = immunization[:date] ? -immunization[:date].to_i : 0`

But that doesn't apply to your changes since they're a mix of strings and integers. This would almost work with group name if you just called `.to_s` on the group name (which would turn the nil to an empty string), but that would sort the empties toward the front.